docs_src/design/adr/security/tbd-authentication-e2ee.md

```diff
@@ -0,0 +1,414 @@
+# EdgeX Microservice Authentication (end-to-end security)
+
+### Submitters
+
+- Bryon Nevis (Intel)
+- Clint Dovholuk (NetFoundry - OpenZiti)
+
+## Change Log
+
+- [proposed](https://github.com/edgexfoundry/edgex-docs/pull/935) (2023-01-28)
+
+## Referenced Use Case(s)
+
+- [Microservice Authentication](https://docs.edgexfoundry.org/3.0/design/ucr/Microservice-Authentication/)
+
+
+## Context
+
+The AS-IS Architecture figure below depicts the current state of
+microservice communication security prior to EdgeX 3.0,
+when security is enabled:
+
+![AS-IS Architecture](tbd-as-is.jpg)
+
+As shown in the diagram,
+many of the foundational services used by EdgeX Foundry
+have already been secured:
+
+* Communication with EdgeX's secret store, as implemented by
+  Hashicorp Vault, is secured over a local HTTP socket with
+  token-based authentication.  An access control list limits access
+  to the keyspace of the key value store.
+
+* Communication with EdgeX's service registry and configuration provider,
+  as implemented by Hashicorp Consul, is secured over a local HTTP
+  socket with token-based authentication,
+  with the token being mediated by Hashicorp Vault.
+  An access control list limits access to the keyspace of
+  the configuration store.
+
+* Communication with EdgeX's default database, Redis, is secured using
+  username/password authentication, with the password stored
+  in Hashicorp Vault.  An access control list limits the commands
+  that clients are allowed to issue to the server.
+
+External access to EdgeX microservices has also been secured.
+EdgeX microservices only bind to local ports,
+and are only exposed externally through a Kong API gateway.
+This gateway is configured to use TLS 1.3,
+using RS256 or ES256 JWT authentication (at the user's discretion).
+All external requests are filtered at the API gateway.
+URL rewriting is used to concentrate microservices
+on a single HTTP-accessible port.
+
+Behind the proxy, it is not possible to verify Kong
+as the origin of local network traffic because mutual-auth TLS
+is not supported in the open source version of Kong.
+Although the Kong JWT plugin will set request headers
+on the backend request that identify the caller,
+there is no mechanism by which Kong can prove to a
+backend service that it was the component that
+performed the authentication step.
+Even though the original JWT passes through the proxy,
+the Kong authentication plugins do not expose
+token introspection endpoints that the backend service
+could use to check token validity independently.
+
+The consequence of having an API gateway that performs
+all microservice authentication is that communication
+between EdgeX microservices running behind the API gateway
+are not authenticated in any way.
+EdgeX microservices are unable to distinguish
+malicious traffic that has evaded the API gateway
+from legitimate microservice traffic.
```

Comment:
Does this need to be updated for the Microservice Auth implementation in 3.0?

Reply:
This entire section got removed when updating the current current state.